tests/integration/helpers/factories/db/invenio_records.py

```diff
@@ -81,25 +81,34 @@ def create_from_kwargs(cls, index=True, has_pid=True, **kwargs):
         instance.record_metadata = super(TestRecordMetadata, cls)\
                 .create_from_kwargs(updated_kwargs)
 
-        if index:
+        if index == 'records-hep':
             instance.es_index_result = es.index(
                 index='records-hep',
                 doc_type='hep',
                 body=instance.record_metadata.json,
                 params={}
             )
             instance.es_refresh_result = es.indices.refresh('records-hep')
+        elif index == 'records-data':
+            instance.es_index_result = es.index(
+                index='records-data',
+                doc_type='data',
+                body=instance.record_metadata.json,
+                params={}
+            )
+            instance.es_refresh_result = es.indices.refresh('records-data')
 
-        if has_pid:
+        if pid_type:
```

Comment:
You have to pass the kwargs around like: https://github.com/inspirehep/inspire-next/blob/master/tests/integration/helpers/factories/db/invenio_oauthclient.py#L147
So:
- remove the `pid_type` from the signature of this method
- remove the if clause at this line
- change the next line to
```
... = TestPersistentIdentifier.create_from_kwargs(
    object_uuid=instance.record_metadata.id,
    pid_value=instance.record_metadata.json.get('control_number'),
    **kwargs).persistent_identifier
```

Reply:
This way we do not need to specify the `pid_type` when you want a `lit` because `TestPersistentIdentifier` defaults to that.